### ISSUE TITLE
JSON routing requests are key order-sensitive

### ISSUE THREAD

**Issue by vkrause** [motis-project/motis#433]
Taking the following JSON request:

```
{
  "destination": {
    "type": "Module",
    "target": "/intermodal"
  },
  "content_type": "IntermodalRoutingRequest",
  "content": {
    "start": {
      "station": {
        "name": "Zürich HB",
        "id": "delfi_000008503000"
      },
      "interval": {
        "begin": 1707317340,
        "end": 1707324540
      },
      "min_connection_count": 5,
      "extend_interval_earlier": true,
      "extend_interval_later": true
    },
    "start_modes": [
      {
        "mode_type": "FootPPR",
        "mode": {
          "search_options": {
            "profile": "default",
            "duration_limit": 900
          }
        }
      }
    ],
    "start_type": "PretripStart",
    "destination_type": "InputStation",
    "destination": {
      "name": "Bern",
      "id": "delfi_000008507000"
    },
    "destination_modes": [
      {
        "mode_type": "FootPPR",
        "mode": {
          "search_options": {
            "profile": "default",
            "duration_limit": 900
          }
        }
      }
    ],
    "search_type": "Accessibility",
    "search_dir": "Forward",
    "router": ""
  }
}
```

and submitting it with the following command works (it gives you actual results today and will complain about being out of the time range in the near future, all of which is considered working).

`curl 'https://europe.motis-project.de/' -X POST -H 'Content-Type: application/json' --data @request.json`

If you however swap the `start_type` and `destination_type` entries (ie. move the entire `start_type` row down one line, keeping the JSON semantically identical) it'll fail with the following error:

```
{
  "_type": "MotisError",
  "error_code": 1,
  "category": "motis::module",
  "reason": "module: unable to parse message"
}
```

This is especially unfortunate for clients using JSON libraries that produce JSON with ordered keys and/or provide no manual control over the key order.

**Comment by felixguendling** [comment]
Hmm. That's bad. I was sure we had this fixed.

The problem is that Flatbuffers (at least the version we're using) requires the "*_type" fields before the actual value and we have [`fix_json.cc`](https://github.com/motis-project/motis/blob/master/base/module/src/fix_json.cc) to fix this issue. Seems like it's not working as intended. We'll have a look. Thank you for the bug report!

**Comment by vkrause** [comment]
That's matching what I had suspected from looking at the code. Now that I have a running version locally and can observe fix_json at work, I think it's actually not the `_type` ordering as such (fix_json is doing that correctly), but `destination` being placed before `start`. That is harder to fix generically at that level though, from what I understand so far.

**Comment by pablohoch** [comment]
This should now be fixed in #435. I haven't tested it extensively, but it works when reordering the start and destination type fields.

**Comment by vkrause** [comment]
Perfect, thank you! Journey queries now work here locally without any ordering hacks/workarounds.
